docs/markdown/Commands.md

```diff
@@ -423,8 +423,20 @@ format should be used. There are currently 3 formats supported:
 - `vscode`: Same as `sh` but without `$VAR` substitution because they do not
   seems to be properly supported by vscode.
 
-{{ devenv_arguments.inc }}
+*Since 1.7.0* `--bin <target>` argument can be used to build and run a given
+executable target. See [`compile`](#compile) command for `target` syntax.
+The remainder of the command line are arguments passed to the target executable:
+```
+meson devenv -C builddir --bin gst-launch-1.0 videotestsrc ! glimagesink
+```
 
+*Since 1.7.0* `run` is an alias of `devenv` with the difference that if `--bin`
+is not specified it defaults to the project's main executable. It is an error if
+more than one executable is installable and `--bin` is not specified or is
+ambiguous. Since an executable is required, `meson run` will never start an
```

Comment:
This is badly worded and difficult to understand. What is "the project's main executable"?

Reply:
Ultimately I would like to add e.g. `meson.set_run_targ(exe)` to let define from meson.build what it the project's run command. For now if there is one and only one installable exe it picks it, otherwise it's an error.

How would you rephrase?

Reply:
If we don't have such an API then we shouldn't write documentation as if we do -- we can always reword the documentation if and when we get such an API.

If you're suggesting to create a guess mode then it should just say that straight out: "--bin is required unless the project only contains a single installable executable".

Reply:
Rephrased the doc now that we don't have --bin in devenv anymore. Is it better?